test.py

```diff
@@ -11,6 +11,46 @@ def g(s):
     assert f_g.functions == (g, f)  # functions exposed in execution order
 
 
+def test_compose_add_compose():
+    def f(s):
+        return s + 'f'
+    def g(s):
+        return s + 'g'
+    f_wrapped = compose(f)
+    g_wrapped = compose(g)
+    f_g_wrapped = f_wrapped + g_wrapped
+
+    assert f_g_wrapped('') == 'gf'  # Remember that it's f(g(...)) so g runs first
+    assert f_g_wrapped.functions == (g, f)  # functions exposed in execution order
+
+
+def test_compose_add_function():
+    def f(s):
+        return s + 'f'
+    def g(s):
+        return s + 'g'
+    f_wrapped = compose(f)
+    f_g_composed = f_wrapped + g
+    assert f_g_composed('') == 'gf'  # Remember that it's f(g(...)) so g runs first
+    assert f_g_composed.functions == (g, f)  # functions exposed in execution order
+
+
+def test_compose_add_neither_compose_nor_function():
+    def f(s):
+        return s + 'f'
+    f_wrapped = compose(f)
+    g = 'string'
+    
+    raised = False
+
+    try:
+        f_g_wrapped = f_wrapped + g
+    except ValueError as err:
+        raised = True
+    finally:
+        assert raised
```

Comment:
```suggestion
    try:
        f_wrapped + g
    except TypeError:
        return
    assert False, '+ should have raised'
```

Because of a bunch of tiny optimizations of various factors:

Simpler (less logical operations, less indirection, less things, less complecting of things together), shorter (so more relevant information fits within the detailed central spot of human vision, so you pay less saccades and human working memory for reading the same code behavior), produces a relevant self-explanatory failure message, not naming/storing things which we're not using as if we're going to use them (so the reader has more immediate local cues that they eliminate or discount the possibility that they are used later), the error type matches what Python normally raises in these cases.

Actually, testing if an error is raised is not the right test - _Python_ is responsible for raising that `TypeError` if both operands return `NotImplemented`. We as operator overloaders are responsible for returning `NotImplemented`. So on further thought I'd go with:

```suggestion
    result = f_wrapped.__add__(g)
    assert result is NotImplemented
```